storefront/test/integration/workarea/storefront/recent_views_integration_test.rb

```diff
@@ -0,0 +1,40 @@
+require 'test_helper'
+
+module Workarea
+  module Storefront
+    class RecentViewsIntegrationTest < Workarea::IntegrationTest
+      include Storefront::IntegrationTest
+
+      def test_show
+        user = create_user
+        category = create_category
+        href = storefront.product_path(product, via: category.id)
+
+        sign_in(user)
+        Metrics::User.save_affinity(
+          id: user.email,
+          action: 'viewed',
+          product_ids: [product.id]
+        )
+        get storefront.recent_views_path(via: category.id)
```

Comment:
Trying to understand, where does the storefront do this? I'm blanking on it.

Reply:
it doesn't, but flow does.

Reply:
to clarify, flow doesn't actually send this param AFAIK, but it might send other params that affect how a product is priced, and `:via` is the only param that I saw that I could actually test for in the view.